### DISCUSSION TITLE
Slicers : Hide items with no data

### DISCUSSION THREAD

**Discussion by lauraearle515** [JanMarvin/openxlsx2#1168] [Ideas]
I use the ```wb_add_slicer``` function quite frequently, but have to go into the output after to adjust the slicer settings to check "Hide items with no data." I would love a way to apply the setting in the code itself so I don't have to go in and change it in the output.

**Comment by JanMarvin** [comment]
Maybe with params `show_missing = TRUE`?

**Reply by lauraearle515** [reply]
Doesn't look like that worked -- tried ```= FALSE``` as well since that seemed like it might be what I want ( I want the values that are greyed out in the slicer since they wouldn't return anything to disappear -- I have multiple slicers so I want choosing something in one slicer to make it so the other slicers only show the possible values to go with the value chosen for the first slicer -- hoping that makes sense lol)

**Reply by JanMarvin** [reply]
Could you attach a screenshot of the dialogue? This could help me understand the issue (but at the moment still at work).

**Reply by lauraearle515** [reply]
Absolutely! This is the code I'm using:

```
library(openxlsx2)

dat <- data.frame(
  var = c("x", "y", "y", "z", "z", "x"),
  speed = c( 4, 4, 7, 7, 8, 9),
  dis = c( 2, 10, 4, 22, 16, 10),
  slicervar1 = c("option1", "option1", "option1", "option2", "option2", "option2" ),
  slicervar2 = c("choice1", "choice1", "choice2", "choice3", "choice3", "choice1")
)

wb<-wb_workbook()
wb$add_worksheet("dat")$add_data(x = dat)
wb$add_worksheet("pivot")
df<- wb_data(wb, 1)

wb$add_pivot_table(df, 2, dims = "A3", slicer = c("slicervar1", "slicervar2"), rows = c("var"), data = "speed", params = list(name = "pivot_1"))$add_slicer(x = df, dims = "D3:E9", slicer = "slicervar1", pivot_table = "pivot_1", params = list(show_missing = TRUE))$add_slicer(x = df, dims = "F3:G9", slicer = "slicervar2", pivot_table = "pivot_1",params = list(show_missing = TRUE))

wb_save(wb, "outputname.xlsx", overwrite = T)

xl_open("outputname.xlsx")
``` 
The setting I'm trying to change is reached by selecting the slicer, hitting slicer in the top ribbon of excel, and then Slicer Settings in the Slicer section. By default, the "Hide items with no data" is unselected as pictured below (this is true with openxlsx2 or just making a pivot/slicer in excel manually). In this example, choosing option1 in the first slicer should make choice3 disappear in the second slicer. Alternatively, choosing choice3 in the second slicer should make option1 disappear. Changing the param ```show_missing``` doesn't seem to effect anything.



![image](https://github.com/user-attachments/assets/878e80d9-bd3f-42e9-b6ac-aae44df04b5d)

As always, thank you so much for your help and if this isn't currently possible, no worries -- just figured I'd ask :)

**Reply by JanMarvin** [reply]
Thanks! This PR should enable what you want to do: https://github.com/JanMarvin/openxlsx2/pull/1169

You can install it as follows: `remotes::install_github("JanMarvin/openxlsx2#1169)`

Let me know if you prefer another name for `hide_no_data_items`.

``` r
library(openxlsx2)

dat <- data.frame(
  var = c("x", "y", "y", "z", "z", "x"),
  speed = c(4, 4, 7, 7, 8, 9),
  dis = c(2, 10, 4, 22, 16, 10),
  slicervar1 = c("option1", "option1", "option1", "option2", "option2", "option2" ),
  slicervar2 = c("choice1", "choice1", "choice2", "choice3", "choice3", "choice1")
)

wb <-wb_workbook()$
  add_worksheet("pivot")$
  add_worksheet("dat")$add_data(x = dat)

df <- wb_data(wb)

wb$
  add_pivot_table(df, sheet = "pivot", dims = "A3", slicer = c("slicervar1", "slicervar2"), rows = c("var"), data = "speed", params = list(name = "pivot_1"))$
  add_slicer(x = df, sheet = "pivot", dims = "D3:E9", slicer = "slicervar1", pivot_table = "pivot_1", param = list(hide_no_data_items = TRUE))$
  add_slicer(x = df, sheet = "pivot", dims = "F3:G9", slicer = "slicervar2", pivot_table = "pivot_1", param = list(hide_no_data_items = TRUE))

if (interactive()) wb$open()
```

**Reply by lauraearle515** [reply]
This did the trick - thank you!! You are the BEST!!

**Reply by JanMarvin** [reply]
Perfect! You know how to keep your developer happy 😄

**Reply by JanMarvin** [reply]
It's now merged into `main`
